README.md

```diff
@@ -1,3 +1,163 @@
+
+The default download location is `~/.minio_runner` directory, and this is configurable:
+
+ ```ruby
+MinioRunner.config.install_dir = '/minio_runner/install/dir'
+```
+
+Alternatively, you can define the path via the `MINIO_RUNNER_INSTALL_DIR` environment variable.
+The environment variable will take precedence.
+
+### Caching minio version
+
+You can set Minio Runner to only look for updates if the previous check
+was longer ago than a specified number of seconds.
+
+```ruby
+MinioRunner.config.cache_time = 86_400 # Default: 86,400 Seconds (24 hours)
+```
+
+Alternatively, you can define the time via the `MINIO_RUNNER_CACHE_TIME` environment variable.
+The environment variable will take precedence.
+
+### Rake tasks
+
+You can run `bundle exec rake -T -a` to see all the rake tasks. The ones specifically related to
+minio runner will be namespaced into minio_runner. 
+
+### Logging
+
+The logging level can be configured for debugging purpose via the `MINIO_RUNNER_LOG_LEVEL` environment variable.
+
+The available values are found in https://ruby-doc.org/stdlib-2.4.0/libdoc/logger/rdoc/Logger/Severity.html.
+
+The minio server will log to the `install_dir` in a `minio.log` file.
+
+## Minio configuration
+
+Only a small subset of minio configuration (defined at https://min.io/docs/minio/linux/reference/minio-server/minio-server.html#environment-variables)
+is supported. The subset of configuration options can be found from running the `list_configurable_env`
+rake task.
+
+All minio configuration can also be specified via `MinioRunner.config`, and anything
+set in this way will override environment variables. Environment variables should
+be in the format `MINIO_RUNNER_MINIO_X`:
+
+```ruby
+MinioRunner.config do |config|
+  config.minio_port = 9000 # MINIO_RUNNER_MINIO_PORT
+  config.minio_console_address = 9001 # MINIO_RUNNER_MINO_CONSOLE_ADDRESS
+  config.minio_domain = 'minio.local' # MINIO_RUNNER_MINIO_DOMAIN
+end
+```
+
+The configuration in ruby will use the exact same names as the environment
+variables for minio.
+
+### Aliases
+
+By default a `local` alias is automatically created via the `mc` tool, which will point
+to `localhost` at the configured `MINIO_RUNNER_MINIO_PORT`. No other aliases are supported
+at this time.
+
+### Buckets
+
+You can specify the buckets that will be created (if they do not exist) when the minio server
+starts using the `MinioRunner.config` call above or using the `MINIO_RUNNER_BUCKETS` environment
+variable with a comma-separated list. Only S3-compatible buckets will be made.
+
+```ruby
+MinioRunner.config.buckets = ["testbucket", "media"]
+
+# MINIO_RUNNER_BUCKETS="testbucket,media"
+```
+
+Buckets will be made public to anonymous users if they are specified in the `public_buckets` configuration,
+which can also be set with the `MINIO_RUNNER_PUBLIC_BUCKETS` environment variable.
+
+### Hosts file
+
+**An important step** that you must manually do yourself is to modify your `/etc/hosts` file to add an
```

Comment:
This is not ideal for the 'spec' use case we're planning for Discourse... Will it be required there? Or can we just use `localhost:port` instead?

Reply:
@davidtaylorhq Unfortunately we can't use `localhost:port` because one of the main parts of using minio is being able to send requests in the format `bucketname.domain.xyz`, which is the virtual host style of requests https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#virtual-hosted-style-access . 

For the spec case in Discourse, I don't forsee there being very many of these specs at all, and mostly it should be run on CI unless someone is specifically working on this stuff. Maybe to avoid people having to mess around with hosts when they don't need to, we segregate those system specs in core and they only run if `ENV["RUN_S3_UPLOAD_SPECS"]` is true or something? Not sure if you saw the core PR but here is the context:

https://github.com/discourse/discourse/pull/22975

AFAIK there is not another way to do this. We could also potentially check in the `/etc/hosts` file when you run `MinioRunner` to prompt people to edit it?